ledger/execution/tx_coinbase.go

```diff
@@ -226,6 +217,15 @@ func grantStakerReward(view *st.StoreView, validatorSet *core.ValidatorSet, guar
 					}
 					stakeAmount := stake.Amount
 					stakeSource := stake.Source
+
+					if blockHeight >= common.HeightSampleStakingReward {
```

Comment:
Do we need to account for the amount stake in the random sampling? e.g. there are only two guardian staker A and B, where A staked 9000 and B staked 1000, then A should be 9 times more likely to be selected than B?